File: caesar_cypher/caesar_cypher.java
import java.util.Scanner;
public class caesar_cypher {
    public static void main(String[] arg){
        Scanner scan = new Scanner(System.in);
        System.out.println("Insira o texto a ser codificado: ");
        String text = scan.nextLine();
        System.out.println("Insira o deslocamento do alfabeto: ");
        int shift = scan.nextInt();
        if(shift>27){
            shift = shift%27;
        }
        int b = text.length();
        int a = 0;
        String c = "\0";
        for(int i = 0; i<b; i++){
            char C = text.charAt(i);
            a = (int)C;
            if (a+shift>=123){
                a += shift-27;
                c +=(char)a;
            }
            else{
                a += shift;
                c +=(char)a;
            }
        }
        System.out.println(c);
    }
}

File: caesar_cypher/caesar_cypher.cpp
#include <iostream>
using namespace std;

int main(){
    string texto;
    int shift;
    int a;
    cout << "Insira o texto a ser codificado: ";
    cin >> texto;
    cout << "Insira o deslocamento do alfabeto: ";
    cin >> shift;
    string K;
    if (shift > 27){
        shift = shift%27;
    }
    else{
        shift = shift;
    }
    int b = texto.length();
    for (int i=0; i<b;++i){
        //cout<< texto[i];
         a = int(texto[i]);
         if (a+shift>=123){
             a += shift-27;
             char newText = a;
             K += newText;
         }
         else{
             a += shift;
             char newText = a;
             K += newText;
         }   
    }
    cout << K;
}